src/BlockchainWriter/DAO.ts

```diff
@@ -32,6 +32,18 @@ export class DAO {
       blockHeight: null,
     })
 
+  readonly purgeStaleTransactions = (currentBlockHeight: number, maxBlockHeightDelta: number) =>
+    this.blockchainWriterCollection.updateMany(
+      {
+        txId: { $ne: null },
+        blockHeight: null,
+        creationBlockHeight: { $lt: currentBlockHeight - maxBlockHeightDelta },
```

Comment:
I feel like this new method has too much knowledge of the logic.

I think this function should just take a block height to compare against instead of doing the math. I think the math should be left in the controller.

Reply:
ack